hack/jenkins/kicbase_auto_build.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+# Copyright 2021 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -x
+
+# Make sure docker is installed and configured
+./hack/jenkins/installers/check_install_docker.sh
+yes|gcloud auth configure-docker
+
+# Setup variables
+now=$(date +%s)
+KV=$(egrep "Version =" pkg/drivers/kic/types.go | cut -d \" -f 2 | cut -d "-" -f 1)
+KIC_REPO=gcr.io/k8s-minikube/kicbase-builds
+export KIC_VERSION=$KV-$now-$ghprbPullId
+export KICBASE_IMAGE_REGISTRIES=$KIC_REPO:$KIC_VERSION
+
+# Let's make sure we have the newest kicbase reference
+curl -L https://github.com/kubernetes/minikube/raw/master/pkg/drivers/kic/types.go --output types-head.go
+HEAD_KIC_TIMESTAMP=$(egrep "Version =" types-head.go | cut -d \" -f 2 | cut -d "-" -f 2)
+CURRENT_KIC_TS=$(egrep "Version =" pkg/drivers/kic/types.go | cut -d \" -f 2 | cut -d "-" -f 2)
+if [[ $HEAD_KIC_TIMESTAMP != v* ]] && [[ $CURRENT_KIC_TS != v* ]]; then
+	diff=$((CURRENT_KIC_TS-HEAD_KIC_TIMESTAMP))
+	if [ $diff -lt 0 ]; then
+		curl -s -H "Authorization: token ${access_token}" \
+		-H "Accept: application/vnd.github.v3+json" \
+		-X POST -d "{\"body\": \"Hi ${ghprbPullAuthorLoginMention}, your kicbase info is out of date. Please rebase.\"}" "https://api.github.com/repos/kubernetes/minikube/issues/$ghprbPullId/comments"
+		exit 1
+	fi
+fi
+rm types-head.go
+
+# Build a new kicbase image
+yes|make push-kic-base-image
```

Comment:
does this also push to docker hub too ? I couldn't pull the docker hub
```
$ docker pull kicbase/stable:v0.0.17-1613614091-10408
Error response from daemon: manifest for kicbase/stable:v0.0.17-1613614091-10408 not found: manifest unknown: manifest unknown

```

Reply:
no, it only pushes to GCR currently, we can add dockerhub fairly easily by adding it to the KICBASE_IMAGE_REGISTRIES env var. this will also complicate what the replacements in types.go needs to be, depending on what repo name we want to use.

we swapped the GCR one from kicbase to kicbase-builds, do we want to do something similar for dockerhub?

Reply:
sure lets do that
lets have change kicbase/stable to kicbase/build
same for github packages
I will add minikube-bot to the docker hub so it has permissions